build.go

```diff
@@ -119,46 +129,60 @@ func build(folder string) error {
 			cmds[0][0] = "perl"
 			cmds[0][1] = "./Configure"
 		} else if runtime.GOOS == "darwin" {
-			if runtime.GOARCH == "arm64" {
-				cmds[0] = append(cmds[0], "darwin64-arm64-cc")
-			} else {
-				cmds[0] = append(cmds[0], "darwin64-x86_64-cc")
-			}
```

Comment:
By removing this, does `openssl` still work on macOS by default regardless of arch currently running in? Or are we adding burden on all users just to help cross-compiling users? If we need some kind of option to help cross-compiling users, we should add it. Similarly, we should document how to cross compile if it is something that is important but not supported with the default options.

Reply:
Sorry, I missed the default OS compiler option.

I was wanting to remove `runtime.GOARCH` for openssl target architecture detection. `GOARCH` is about GO compile target, rather than openssl. If `GOARCH` is different from build env, `go run build.go` cannot be run.

I replace it using `uname -m` to detect default target architecture